### ISSUE TITLE
Feature Request: Support for variable names with special characters

### ISSUE THREAD

**Issue by theomails** [fasseg/exp4j#65]
Could you add support for expressions where the variable names have special characters?

For example "[Salary.BASIC] \* 0.40" with the variable as [Salary.BASIC] 

If the value of [Salary.BASIC] is $1000.0, the expression result should be $400.0

**Comment by fasseg** [comment]
You mean that the square brackets `[` and `]` should be part of the variable name?

**Comment by theomails** [comment]
I thought of the square brackets as similar to single quotes. They are optional for normal variables and mandatory for fields with special characters.

**Comment by fasseg** [comment]
Okay so bad news first: You can't use square brackets as part of the variable name, since these have different semantics in a mathematical context i.e. grouping of operations. You should be able to use `Salary.Basic` as a variable name though out of the box in exp4j.
 Square brackets are not possible since it does create disambiguities, if you have e.g. a variable `x` and a variable `[x]`, the system has no way to identify which variable is meant by the expression `2*[x]`.

**Comment by theomails** [comment]
I tried `Salary.Basic` as the variable name. It generates a runtime exception: 

```
double result3 = new ExpressionBuilder("percent( Salary.Basic , 40)")
		        .function(percent)
		        .variables("Salary.Basic")
		        .build()
		        .setVariable("Salary.Basic", 30_000.0)
		        .evaluate();
```

> Unknown function or variable 'Salary' at pos 9 in expression 'percent( Salary.Basic , 40)'

**Comment by fasseg** [comment]
Pretty sure this is a bug. I will check the sources and come back to you..

**Comment by fasseg** [comment]
I pushed an update to master. Can you please verify that this fix works for you?

**Comment by theomails** [comment]
Its working now. I'm able to use Salary.Basic as a single variable name.

**Comment by fasseg** [comment]
released in 0.4.8
